### ISSUE TITLE
Different behavior of error handlers

### ISSUE THREAD

**Issue by Danstiv** [pgjones/quart#121]
Hello, i want to change default error pages of quart. Here's an flask example.
```
import flask
from werkzeug.exceptions import HTTPException
import json
app = flask.Flask(__name__)
@app.route('/')
def index():
	3/0
def error_handler(exc):
	return flask.Response(json.dumps({'status': 0, 'description': f'HTTP error {exc.code}'}), status=exc.code, headers={'content-type': 'application/json'})
app.register_error_handler(HTTPException, error_handler)
app.run()
```
Testing:
```
curl -i http://localhost:5000/ff
HTTP/1.0 404 NOT FOUND
content-type: application/json
Content-Length: 46
Server: Werkzeug/1.0.1 Python/3.8.2
Date: Thu, 25 Mar 2021 08:26:51 GMT

{"status": 0, "description": "HTTP error 404"}
curl -i http://localhost:5000
HTTP/1.0 500 INTERNAL SERVER ERROR
content-type: application/json
Content-Length: 46
Server: Werkzeug/1.0.1 Python/3.8.2
Date: Thu, 25 Mar 2021 08:24:18 GMT

{"status": 0, "description": "HTTP error 500"}
```
everything works well.
quart example:
```
import quart
from quart.exceptions import HTTPException
import json
app = quart.Quart(__name__)
@app.route('/')
async def index():
	3/0
async def error_handler(exc):
	return quart.Response(json.dumps({'status': 0, 'description': f'HTTP error {exc.status_code}'}), status=exc.status_code, headers={'content-type': 'application/json'})
app.register_error_handler(HTTPException, error_handler)
app.run()
```
testing:
```
curl -i http://localhost:5000/ff
HTTP/1.1 404
content-type: application/json
content-length: 46
date: Thu, 25 Mar 2021 08:29:27 GMT
server: hypercorn-h11

{"status": 0, "description": "HTTP error 404"}
curl -i http://localhost:5000/
HTTP/1.1 500
content-type: text/html; charset=utf-8
content-length: 15173
date: Thu, 25 Mar 2021 08:30:54 GMT
server: hypercorn-h11


<style>
pre {
  margin: 0;
}

.traceback, .locals {
  display: table;
  width: 100%;
  margin: 5px;
}

.traceback>div, .locals>div {
  display: table-row;
}

.traceback>div>div, .locals>div>div {
  display: table-cell;
}

.locals>div>div {
  border-top: 1px solid lightgrey;
}

.header {
  background-color: #ececec;
  margin-bottom: 5px;
}

.highlight {
  background-color: #ececec;
...
<h1>AttributeError <span>'ZeroDivisionError' object has no attribute 'status_code'</span></h1>
<ul>

    <li>
      <div class="header">
        File <span class="info">quart_example.py</span>,
        line <span class="info">9</span>, in
      </div>
      <div class="traceback">

          <div class="highlight">
...
```
Why my HTTPException handler received completely different exception, and How can I get the flask behavior?

**Comment by Danstiv** [comment]
Version from "master" with werkzeug exceptions also have this problem.

**Comment by pgjones** [comment]
I think this is now fixed in master, see ed248c66d3b79402b0ff3bd9c4c0ab28ddfcdf4a, and cb3b15740d435007d17cf36a080ba8063986b68e.

**Comment by Danstiv** [comment]
```
curl -i http://localhost:5000/
HTTP/1.1 500
content-type: application/json
content-length: 46
date: Mon, 29 Mar 2021 04:30:07 GMT
server: hypercorn-h11

{"status": 0, "description": "HTTP error 500"}
```
Handler with exception:
```
async def error_handler(exc):
	3/0
```
Testing with new handler: 
```
curl -i http://localhost:5000/ff
HTTP/1.1 500
content-length: 0
connection: close
date: Mon, 29 Mar 2021 04:36:47 GMT
server: hypercorn-h11


```
Exceptions now handled correctly, also the exception from the exception handler is not sent to the user.
The difference from flask is that it falls back to the standard 500 response if an exception occurs in exception handler.
Quart just closes the connection, but I do not think that this difference can affect anything.
Current behavior is convenient for me. It works for my use-case.
